### PR TITLE
feat(libs): move fxa-shared/l10n to libs/shared/l10n

### PR DESCRIPTION
**Pull request by StaberindeZA**
## Because

* We want to use functions from fxa-shared/l10n in new libs/* libraries including libs/shared/contentful.

## This pull request

* Moves fxa-shared/l10n to libs/shared/l10n
* Updates all references in packages/*

## Issue that this pull request solves

Closes: #FXA-8228

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

## Additional notes

* There were a couple of references that were a little trickier to update. I filed a follow up ticket, FXA-8413, to resolve these.

### CODE REVIEW

File: packages/fxa-react/configs/rescripts.js [hunk 1]
```diff
@@ -58,6 +58,18 @@ const permitAdditionalJSImports = (config) => {
...
+  config.resolve.alias = {
+    ...config.resolve.alias,
+    '@fxa/shared/l10n': resolve(
+      __dirname,
+      '../../fxa-react/dist/libs/shared/l10n/src/index.js'
+    ),
+  };
+
+  config.resolve.fallback = { fs: false, path: false };
+
```
Comment (StaberindeZA) [inline packages/fxa-react/configs/rescripts.js:72]: This is really not pretty, but not sure how to get it working otherwise.

Comment (StaberindeZA) [reply packages/fxa-react/configs/rescripts.js]: Reworking this a bit, and this comment is no longer applicable.

File: packages/fxa-shared/l10n/parseAcceptLanguage.ts [hunk 2]
```diff
@@ -1,9 +1,9 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-import { negotiateLanguages } from '@fluent/langneg';
-import availableLocales from './supportedLanguages.json';
-import { EN_GB_LOCALES } from './otherLanguages';
+// import { negotiateLanguages } from '@fluent/langneg';
```
Comment (StaberindeZA) [inline packages/fxa-shared/l10n/parseAcceptLanguage.ts:4]: Ah whoops! I forgot to delete this. Will delete it before merging.

**Comment by bcolsson** [comment]
@StaberindeZA - this will also affect our string extraction [here](https://github.com/mozilla/fxa-content-server-l10n/blob/c567e49b46c05e9f092130b566a73fa78367ef80/.github/workflows/l10n_extract.yaml#L51C18-L51C18). Could you also make a PR with that update?

**Comment by StaberindeZA** [comment]
> @StaberindeZA - this will also affect our string extraction [here](https://github.com/mozilla/fxa-content-server-l10n/blob/c567e49b46c05e9f092130b566a73fa78367ef80/.github/workflows/l10n_extract.yaml#L51C18-L51C18). Could you also make a PR with that update?

Ah thank you @bcolsson. I've put up the following PR [mozilla/fxa-content-server-l10n/pull/762](https://github.com/mozilla/fxa-content-server-l10n/pull/762). I tested it locally and everything seems good. Could you please review if you have a second?

**Review by bcolsson — APPROVED** [review]

**Review by IvoJP — APPROVED** [review]

**Pull request merged**